conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt

```diff
@@ -309,57 +286,40 @@ class ExternalDatabasePermissioner(
+                columnAclKeyToName = columnAclKeyToName,
                 permissions = allViewPermissions
-
         )
     }
 
     private fun initializePermissionSetForExternalTable(
             hikariDataSource: HikariDataSource,
-            tableSchema: String,
-            tableName: String,
-            columns: Set<ExternalColumn>,
+            columnAclKeyToName: Map<AclKey, String>,
             permissions: Set<Permission>
     ) {
-        val targetsToColumns = mutableMapOf<AccessTarget, ExternalColumn>()
-
-        val targets = columns.flatMapTo(mutableSetOf()) { column ->
-            permissions.map { permission ->
-                val at = AccessTarget.forPermissionOnTarget(permission, column.tableId, column.id)
-                targetsToColumns[at] = column
-                at
-            }
-        }
-
-        PostgresRoles.getOrCreatePermissionRolesAsync(
-                externalRoleNames,
-                targets,
-                hikariDataSource
-        ).thenApplyAsync { targetToRoleNames ->
-            targetToRoleNames.map { (target, roleName) ->
-                val column = targetsToColumns.getValue(target)
-                val pgPermissions = olToPostgres.getValue(target.permission)
-                grantPermissionsOnColumnsOnTableToRoleSql(
-                        pgPermissions,
-                        ApiHelpers.dbQuote(column.name),
-                        tableSchema,
-                        tableName,
-                        roleName.toString()
-                )
-            }
-        }.thenAccept { sqls ->
-            hikariDataSource.connection.use { conn ->
-                conn.createStatement().use { stmt ->
-                    sqls.forEach { sql ->
-                        stmt.addBatch(sql)
+        // insert column name into external_permission_roles as role_id, needs more thinking
+        // actually, why do we even need external_permission_roles? isn't information_schema.column_privileges enough?
+        hikariDataSource.connection.use { conn ->
+            conn.createStatement().use { stmt ->
+                permissions.forEach { permission ->
+                    columnAclKeyToName.filterNot {
+                        externalRoleNames.containsKey(AccessTarget(it.key, permission))
+                    }.forEach { 
+                        olToPostgres.getValue(permission).forEach { privilege -> 
+                            val newRoleId = UUID.randomUUID()
+                            stmt.addBatch("""
+                                INSERT INTO ${HazelcastMap.EXTERNAL_PERMISSION_ROLES.name}
+                                VALUES ('{\"${it.key}\"}'::uuid[], \"$privilege\", ${ApiHelpers.dbQuote(it.value)}, \"${newRoleId}\"::uuid) ON CONFLICT DO NOTHING
+                            """.trimIndent())
+                        }
                     }
-                    stmt.executeBatch()
                 }
+                stmt.executeBatch()
```

Comment:
Yeah, currently all `GRANT` commands are done through respective `update...` functions. 

The reason why this has a `GRANT` command is because `initialize...` creates, for each of the column-permission role_id, a role and grants the required permission, and `update...` adds the requested ace (roles, users) to the column-permission role. Since we now directly grant permissions to the ace, I removed all `GRANT` operations from `initialize...` functions and placed them in `update...` instead. This can be changed of course.